javascript/es6.js

```diff
@@ -0,0 +1,32 @@
+export default () => {
+    const arr =[1, 2, 3];
+    arr.push(4);
+    console.log(arr);
+    try {
+        arr = 3;
+    }catch(err){
+        console.log(err);
+    }
+
+    class Person {
+        constructor(fname, lname){
+            this.fname = fname;
+            this.lname = lname;
+        }
+    }
+    function person(ime, prezime){
+        this.ime = ime;
+        this.prezime = prezime;
+    }
+    var a = new person("ime", "prezime");
+    console.log(a);
+    var p = new Person("ime", "prezime");
+    console.log(p);
+
+    let arrow = (ime, prezime) => {
+        this.ime = ime;
+        this.prezime = prezime;
+    }
+    //undefined
+    console.log(arrow.ime);
```

Comment:
ovdje ce biti `undefined`  zato sto bi treba o prvo pozvati funkciju prije svega , gdje ces dodjeliti `this` ime i prezime. Mejdutim kad i to odradis i console.log odradis opet ces dobiti `undefined` ali pogledaj sta ce `this.ime` ili `window.ime` imati za vrijednost

Resurs vezano za scope i this https://scotch.io/tutorials/understanding-scope-in-javascript